drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -126,7 +126,11 @@ static void nrf5_get_eui64(uint8_t *mac)
 	mac[index++] = (IEEE802154_NRF5_VENDOR_OUI >> 8) & 0xff;
 	mac[index++] = IEEE802154_NRF5_VENDOR_OUI & 0xff;
 
-#if defined(CONFIG_TRUSTED_EXECUTION_NONSECURE) && defined(NRF_FICR_S)
+#if defined(NRF54H_SERIES)
+	/* Can't access SICR with device id on a radio core. Use BLE.ADDR. */
+	deviceid[0] = NRF_FICR->BLE.ADDR[0];
+	deviceid[1] = NRF_FICR->BLE.ADDR[1];
```

Comment:
nit: how about using Zephyr's `hwinfo_get_device_id()` instead of these ifdefs?
Of course, can be addressed later. It's just an idea

Reply:
Thanks for this hint. Current implementation of `hwinfo_get_device_id` for nrf54h20 is non-functional. see https://github.com/zephyrproject-rtos/zephyr/blob/8c8e2f6243f147953ce3fadb999c4e2a5faf4f5e/drivers/hwinfo/hwinfo_nrf.c#L25 That's the reason the workaround is ever needed.

Reply:
+ We are not sure if we want to add a dependency on the hwinfo module.

Reply:
> +We are not sure if we want to add a dependency on the hwinfo module.

I don't see a problem with this dependency in the Zephyr-specific code. It's easier to maintain than dependencies to multiple SoCs.

But again, it's just a hint. If you estimate it's easier for you to maintain SoC-specific defines than to depend on a Zephyr module, it's ok for me.